File: java/javabasico/src/main/java/co/edu/sena/javabasico/desiciones/Cascada.java
package co.edu.sena.javabasico.desiciones;

import java.util.Scanner;

public class Cascada {

    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        int c;
        c = teclado.nextInt();

        if (c == 1) {
            System.out.println("uno");
        } else {
            if (c == 2) {
                System.out.println("dos");
            } else {
                if (c == 3) {
                    System.out.println("tres");
                } else {
                    System.out.println("no se que numero sera");
                }
            }
        }

        // mejor
        if (c == 1) {
            System.out.println("uno");
        } else if (c == 2) {
            System.out.println("uno");
        } else if (c == 3) {
            System.out.println("uno");
        } else if (c == 4) {
            System.out.println("uno");
        } else {
            System.out.println("uno");
        }
    }

}

File: typescript/basic/src/ejemplos/decisiones/cascada.ts


let varc: number;
varc = parseInt(prompt("digite un numero"));

if (varc == 1) {
    console.log("uno");
} else {
    if (varc == 2) {
        console.log("dos");
    } else {
        if (varc == 3) {
            console.log("tres");
        } else {
            console.log("no se que numero sera");
        }
    }
}

// mejor
if (varc == 1) {
    console.log("uno");
} else if (varc == 2) {
    console.log("dos");
} else if (varc == 3) {
    console.log("tres");
} else if (varc == 4) {
    console.log("cuatro");
} else {
    console.log("cinco");
}